app/models/section.rb

```diff
@@ -22,6 +23,32 @@ def method_missing(m, *args, &block)
       super
     end
   end 
+  
+  def self.import(file, section)
+    if File.extname(file.tempfile) == ".csv"
+      # Parse CSV and get into 1D array
+      csv_text = File.read(file.tempfile, :encoding => 'utf-8')
+      emailsCSV = CSV.parse(csv_text)
+      emailsCSV = emailsCSV.flatten
+
+      @section = Section.find_by_id(section)
+      
+      emailsCSV.each do |csv_email|
+        emailObj = Email.find_by_email(csv_email)
+        puts "Conditional logic: #{emailObj}"
+        if emailObj
```

Comment:
You need a check to see if its not already in the section too. Something like:
`sectionEmail = @section.emails.find_by_email(csv_email)`
`unless sectionEmail`
`    @section.emails << emailObj`  
`end`
`...`